datajunction-server/datajunction_server/construction/build_v2.py

```diff
@@ -297,9 +297,12 @@ def build_preaggregate_query(
     # Find all required GROUP BY columns based on each of the measure's aggregation rules.
     # If the measure supports full aggregation, there are no required group-by columns, but if it
     # supports limited aggregation, we need to aggregate to the specified level.
+    dim_colum_name_to_alias = {
+        str(col.name): str(col.alias) or str(col.name) for col in dimensional_columns
```

Comment:
I think you can replace this with `col.identifier()` or `col.name.identifier()`. So maybe something like:
```
        col.name.identifier(): (col.alias or col.name).identifier() for col in dimensional_columns
```